spark/src/main/scala/org/apache/spark/sql/delta/Snapshot.scala

```diff
@@ -93,6 +95,26 @@ class Snapshot(
 
   override def columnMappingMode: DeltaColumnMappingMode = metadata.columnMappingMode
 
+  /**
+   * Returns the timestamp of the latest commit of this snapshot.
+   * For an uninitialized snapshot, this returns -1.
+   *
+   * When InCommitTimestampTableFeature is enabled, the timestamp
+   * is retrieved from the CommitInfo of the latest commit which
+   * can result in an IO operation.
+   */
+  def timestamp: Long = {
+    if (DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.fromMetaData(metadata)) {
+      inCommitTimestampOpt.getOrElse {
+        val commitInfoOpt = DeltaHistoryManager.getCommitInfoOpt(
+          deltaLog.store, deltaLog.logPath, version, deltaLog.newDeltaHadoopConf())
+        CommitInfo.getRequiredInCommitTimestamp(commitInfoOpt, version)
```

Comment:
This method can throw an exception when the commitInfo doesn't have inCommitTimestamp or when commitInfo itself is not present?

In such a scenario, snapshot.timestamp will fail. Is this intentional?

Reply:
This is intentional. Such a thing happening is indicative of a writer not following the protocol. In my opinion, we should surface this issue as soon as we see it to prevent further breakage.